### ISSUE TITLE
jsx-no-undef does not support dynamic JSX tags using `this`

### ISSUE THREAD

**Issue by darrenscerri** [yannickcr/eslint-plugin-react#489]
```js
class Component extends React.Component {
  render() {
    return (
      <this.props.tag>Test</this.props.tag>
    );
  }
}
```

Eslint is complaining:

```react/jsx-no-undef 'this' is not defined```
